### DISCUSSION TITLE
Use docker-compose.yaml in go testify suite

### DISCUSSION THREAD

**Discussion by rashmy-dev** [testcontainers/testcontainers-go#2841] [Q&A]
Hi,
I have a docker-compose.yaml file with a list of containers that needs to be started before running the e2e tests.
From the documentation here - https://golang.testcontainers.org/features/docker_compose/, it not clear to me how to use the new "ComposeStack".

In my testify suite, I do the following -
```code
type Mye2eTestSuite struct {
	suite.Suite
}
func (suite *Mye2eTestSuite) SetupSuite() {
	compose, err := compose.NewDockerComposeWith(compose.ComposeStackFiles{"../docker-compose.yml"})
	suite.Require().NoError(err, "error starting test containers")
}

func (suite *Mye2eTestSuite) TearDownSuite() {
}
```

The 'compose' returned from NewDockerComposeWith() is of the type *dockerCompose (which as I understand is not exposed and deprecated).
How do I use the new ComposeStack to start the containers in docker-compose.yaml and then stop the containers in TearDownSuite() function?

Thanks!

**Comment by mdelapenya** [comment]
Hi, we have this example repo: https://github.com/mdelapenya/testcontainers-go-compose

Could you take a look?

**Reply by rashmy-dev** [reply]
Yes, I had already seen this example. My question is how to access 'compose' to bring in down in the testify suite.TearDownSuite() function.

**Reply by mdelapenya** [reply]
Ok, I'll try to create an example in that repo using testify's suite structs. Will ping you once I have it working

**Reply by rashmy-dev** [reply]
Thank you!

**Reply by rashmy-dev** [reply]
@mdelapenya 
Just checking to see if you are able share the solution to access the created docker containers and bring it down in the testify suite.TearDownSuite() function.

Thanks.

**Reply by mdelapenya** [reply]
Hey @rashmy-dev I could not find time for it, although I found this article that could help you out: https://www.dolthub.com/blog/2024-02-05-testcontainers-for-go-with-dolt/. Please look at the `customer/clone_suite_test.go` file described in there.

**Reply by rashmy-dev** [reply]
Thank you for the link. The blog shows the example of manually creating each container in the setup function and then manually shutting down each container in teardown function. My need is to create containers available in a `docker-compose.yaml` file and bring down the same.

The key issue is, I can create the containers like this -
```
type Mye2eTestSuite struct {
	suite.Suite
}
func (suite *Mye2eTestSuite) SetupSuite() {
	compose, err := compose.NewDockerComposeWith(compose.ComposeStackFiles{"../docker-compose.yml"})
	suite.Require().NoError(err, "error starting test containers")
}

func (suite *Mye2eTestSuite) TearDownSuite() {
}
````

But I cannot store a reference to 'compose' returned from NewDockerComposeWith() and later invoke shutdown in TearDownSuite(). The 'compose' is of type dockerCompose, which is not a exposed type and is deprecated. The documentation asks to use ComposeStack but it is not clear to me how.

Thanks,
Rashmy

**Reply by mdelapenya** [reply]
Yeah, you're totally right. @stevenh I'll add a note in [our project board](https://github.com/orgs/testcontainers/projects/13) to enable this, thanks!

**Reply by mdelapenya** [reply]
Issue created here: https://github.com/testcontainers/testcontainers-go/issues/2907, not sure if you would like to work on it
